synthesis/codon/codon_test.go

```diff
@@ -388,3 +389,166 @@ func TestCapitalizationRegression(t *testing.T) {
+							{Triplet: "GCA", Weight: 1}, {Triplet: "GCG", Weight: 1},
+						},
+					},
+					{
+						Letter: "D",
+						Codons: []Codon{{Triplet: "GAT", Weight: 1}, {Triplet: "GAC", Weight: 1}},
+					},
+					{
+						Letter: "S",
+						Codons: []Codon{
+							{Triplet: "TCT", Weight: 1}, {Triplet: "TCC", Weight: 1},
+							{Triplet: "TCA", Weight: 1}, {Triplet: "TCG", Weight: 1},
+							{Triplet: "AGT", Weight: 1}, {Triplet: "AGC", Weight: 1},
+						},
+					},
+					{
+						Letter: "C",
+						Codons: []Codon{{Triplet: "TGT", Weight: 1}, {Triplet: "TGC", Weight: 1}},
+					},
+					{
+						Letter: "R",
+						Codons: []Codon{
+							{Triplet: "CGT", Weight: 1}, {Triplet: "CGC", Weight: 1},
+							{Triplet: "CGA", Weight: 1}, {Triplet: "CGG", Weight: 1},
+							{Triplet: "AGA", Weight: 1}, {Triplet: "AGG", Weight: 1},
+						},
+					},
+					{
+						Letter: "N",
+						Codons: []Codon{{Triplet: "AAT", Weight: 1}, {Triplet: "AAC", Weight: 1}},
+					},
+					{
+						Letter: "V",
+						Codons: []Codon{
+							{Triplet: "GTT", Weight: 1}, {Triplet: "GTC", Weight: 1},
+							{Triplet: "GTA", Weight: 1}, {Triplet: "GTG", Weight: 1},
+						},
+					},
+					{
+						Letter: "F",
+						Codons: []Codon{{Triplet: "TTT", Weight: 1}, {Triplet: "TTC", Weight: 1}},
+					},
+					{
+						Letter: "L",
+						Codons: []Codon{
+							{Triplet: "TTA", Weight: 1}, {Triplet: "TTG", Weight: 1},
+							{Triplet: "CTT", Weight: 1}, {Triplet: "CTC", Weight: 1},
+							{Triplet: "CTA", Weight: 1}, {Triplet: "CTG", Weight: 1},
+						},
+					},
+					{
+						Letter: "P",
+						Codons: []Codon{
+							{Triplet: "CCT", Weight: 1}, {Triplet: "CCC", Weight: 1},
+							{Triplet: "CCA", Weight: 1}, {Triplet: "CCG", Weight: 1},
+						},
+					},
+					{Letter: "W", Codons: []Codon{{Triplet: "TGG", Weight: 1}}},
+					{
+						Letter: "H",
+						Codons: []Codon{{Triplet: "CAT", Weight: 1}, {Triplet: "CAC", Weight: 1}},
+					},
+					{
+						Letter: "T",
+						Codons: []Codon{
+							{Triplet: "ACT", Weight: 1}, {Triplet: "ACC", Weight: 1},
+							{Triplet: "ACA", Weight: 1}, {Triplet: "ACG", Weight: 1},
+						},
+					},
+					{
+						Letter: "K",
+						Codons: []Codon{{Triplet: "AAA", Weight: 1}, {Triplet: "AAG", Weight: 1}},
+					},
+					{
+						Letter: "E",
+						Codons: []Codon{{Triplet: "GAA", Weight: 1}, {Triplet: "GAG", Weight: 1}},
+					},
+					{
+						Letter: "G",
+						Codons: []Codon{
+							{Triplet: "GGT", Weight: 1}, {Triplet: "GGC", Weight: 1},
+							{Triplet: "GGA", Weight: 1}, {Triplet: "GGG", Weight: 1},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		var tt = tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			got := generateCodonTable(tt.aminoAcids, tt.starts)
+
+			if !cmp.Equal(got, tt.wantCodonTable, sortCodons, sortAminoAcids, sortStrings) {
+				t.Errorf("got and tt.wantCodonTable didn't match %s", cmp.Diff(got, tt.wantCodonTable))
+			}
```

Comment:
Use the same `if diff := cmp.Diff(); diff != "" {}` syntax I used elsewhere.

Reply:
is there a particular reason?

Reply:
More of a style choice than anything, though I guess it reduces the number of calls to the `cmp` package. Should have marked this as a nit!